### ISSUE TITLE
Feature Request: allow DDL statements to run for unlimited time

### ISSUE THREAD

**Issue by shlomi-noach** [vitessio/vitess#16710]
### Feature Description

The `vttablet` variable `--queryserver-config-query-timeout` sets a timeout on all statements served in `TabletServer`/`QueryExecutor`. It makes sense for most queries to have some reasonable timeouts.

However, for DDL statements this is problematic, and even wrong. Consider the following arguments:
- Some DDLs like `ALTER TABLE` or `OPTIMIZE TABLE` can runs for hours and days, well beyond any reasonable `queryserver-config-query-timeout` setting.
- Say we do run such a long running DDL. The way the timeout is implemented is like so:
  1. Using a `context.WithTimeout`
  2. Check for timeout here:
    https://github.com/vitessio/vitess/blob/1ed18d04f3ecd0b88722a36b7aea5bacafbd628e/go/vt/vttablet/tabletserver/connpool/dbconn.go#L188-L194

  3. Fast forward a few function calls, issue a `kill query` statement here:
    https://github.com/vitessio/vitess/blob/1ed18d04f3ecd0b88722a36b7aea5bacafbd628e/go/vt/vttablet/tabletserver/connpool/dbconn.go#L520-L525

The problem is that `KILL QUERY` does not work reliably with long running DDLs. The good scenario is that MySQL does rollback the `ALTER TABLE` statement, but it can and will take a proportionate time to the timeframe the statement ran before beign killed. So from the moment of `KILL QUERY` it may yet take hours for the original query to return/complete, which leaves the killing code waiting still.

To summarize:
- DDLs have very different timeout requirements compared with DML
- Timing out a DDL may not work as expected.

In light of that, I'm suggesting DDL should run without timeout at all.

It should go without saying that we advocate the use of Online DDL in vitess.

### Use Case(s)

Being able to protect against normal queries unreasonable runtime, while still allowing long running DDLs.
